include/bmx280.h

```diff
@@ -16,24 +16,42 @@ extern "C" {
 
 #include <stdint.h>
 #include <limits.h>
-#include "driver/i2c.h"
 #include "sdkconfig.h"
 
+#include "bmx280_bits.h"
+#if !(CONFIG_USE_I2C_NEW_DEVICE)
+#include "driver/i2c.h"
+#else
+#include "driver/i2c_master.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#endif
+
 #define BMXAPI extern
 
 /**
  * Anonymous structure to driver settings.
  */
 typedef struct bmx280_t bmx280_t;
 
-#include "bmx280_bits.h"
 
+
+
+#if !(CONFIG_USE_I2C_NEW_DEVICE)
 /**
  * Create an instance of the BMX280 driver.
  * @param port The I2C port to use.
  * @return A non-null pointer to the driver structure on success.
  */
 BMXAPI bmx280_t* bmx280_create(i2c_port_t port);
+#else
+/**
+ * Create an instance of the BMX280 driver.
+ * @param bus_handle The I2C master handle via port.
+ * @return A non-null pointer to the driver structure on success.
+ */
+BMXAPI bmx280_t* bmx280_create(i2c_master_bus_handle_t bus_handle);
```

Comment:
I think the review I made in PR #4 applies here as well. I would rather not create a situation where people can accidentally compile the BMX280 driver with the incorrect I2C driver mode. I suggest you refactor this like so:

```c
#include <assert.h>

#if CONFIG_USE_I2C_NEW_DEVICE
// Your new entry point.
BMXAPI bmx280_t bmx280_create_master(i2c_master_dev_handle_t dev_handle);
#define bmx280_create_legacy(port) static_assert(0, "You have the wrong driver configuration for using the legacy I2C driver.")
#define bmx280_create(port) bmx280_create_legacy(port)
#else
// The original entry point but renamed.
BMXAPI bmx280_t bmx280_create_legacy(i2c_port_t port);
// legacy define for existing code bases
#define bmx280_create(port) (bmx280_create_legacy((port)))
#define bmx280_create_master(port) static_assert(0, "You have the wrong driver configuration for using the new I2C master driver.")
#endif
```

It brings up the C language version from C99 to C11 but I think this is important for legacy users.